qemu/tests/timerdevice_change_guest_clksource.py

```diff
@@ -41,134 +32,41 @@ def verify_guest_clock_source(session, expected):
                           logging.info)
-    cmd = "cat /sys/devices/system/clocksource/"
-    cmd += "clocksource0/current_clocksource"
-    if "kvm-clock" not in session.cmd(cmd):
-        grub_file = params.get("grub_file", "/boot/grub2/grub.cfg")
-        if "clocksource=" not in session.cmd("cat %s" % grub_file):
-            test.fail("Guest didn't use 'kvm-clock' clocksource")
-
-        error_context.context("Shutdown guest")
-        vm.destroy()
-        env.unregister_vm(vm.name)
+    cur_clk = params["cur_clk"]
+    if "kvm-clock" not in session.cmd(cur_clk):
         error_context.context("Update guest kernel cli to kvm-clock",
                               logging.info)
-        image_filename = storage.get_image_filename(params,
-                                                    data_dir.get_data_dir())
-        kernel_cfg_pattern = params.get("kernel_cfg_pos_reg",
-                                        r".*vmlinuz-\d+.*")
-
-        disk_obj = utils_disk.GuestFSModiDisk(image_filename)
-        kernel_cfg_original = disk_obj.read_file(grub_file)
-        try:
-            logging.warn("Update the first kernel entry to"
-                         " kvm-clock only")
-            kernel_cfg = re.findall(kernel_cfg_pattern,
-                                    kernel_cfg_original)[0]
-        except IndexError as detail:
-            test.error("Couldn't find the kernel config, regex"
-                       " pattern is '%s', detail: '%s'" %
-                       (kernel_cfg_pattern, detail))
-
-        if "clocksource=" in kernel_cfg:
-            kernel_cfg_new = re.sub(r"clocksource=[a-z\- ]+", " ", kernel_cfg)
-            disk_obj.replace_image_file_content(grub_file, kernel_cfg,
-                                                kernel_cfg_new)
-
-        error_context.context("Boot the guest", logging.info)
-        vm_name = params["main_vm"]
-        cpu_model_flags = params.get("cpu_model_flags")
-        params["cpu_model_flags"] = cpu_model_flags + ",-kvmclock"
-        env_process.preprocess_vm(test, params, env, vm_name)
-        vm = env.get_vm(vm_name)
-        vm.verify_alive()
-        session = vm.wait_for_login(timeout=timeout)
+        grubby_args = params["grubby_args"]
+        session.cmd(grubby_args % "kvm-clock")
+        error_context.context("Reboot the guest", logging.info)
+        session = vm.reboot(session)
+        verify_guest_clock_source(session, "kvm-clock")
 
     error_context.context("Check the available clocksource in guest",
                           logging.info)
-    cmd = "cat /sys/devices/system/clocksource/"
-    cmd += "clocksource0/available_clocksource"
+    avl_clk = params["avl_clk"]
     try:
-        available_clksrc_list = session.cmd(cmd).splitlines()[-1].split()
+        available_clksrc_list = session.cmd(avl_clk).splitlines()[-1].split()
         available_clksrc_list = [_.strip() for _ in available_clksrc_list]
```

Comment:
I think "available_clksrc_list = session.cmd(avl_clk).split()" is enough for above two lines.